src/System.Collections/tests/Resources/System.Collections.Tests.rd.xml

```diff
@@ -1,7 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
 <Directives xmlns="http://schemas.microsoft.com/netfx/2013/01/metadata">
-  <Library>
+  <Library Name="System.Collections.Tests">
+    <Assembly Name="*Application*" DoNotInline="true" DoNotOptimize="true" />
```

Comment:
> @safern Where do we track the codegen bug? I can't find any mention of it in the pull request that added it.
I did a sweep of the codebase for unaccounted codegen workarounds (e.g. #22475) but I forgot about this sneaky trick. We really should stick to the <ILCBuildType Condition="'$(TargetGroup)' == 'uap'">chk</ILCBuildType> pattern for this.

This bug is tracked in VSTS, bug number: 472947

Kirstie Lee from the codegen bug is working on getting that bug fixed. However we don't have a corefx tracking bug, but I am working with her to get it fixed. @ViktorHofer feel free to delete this file and add `<ILCBuildType>chk</ILCBuildType>` to the csproj. If you want to merge this ASAP let me know and I can take care of doing this update.